projects/driver_controls/src/magnetic_brake_event_generator.c

```diff
@@ -0,0 +1,138 @@
+#include <limits.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include <stdbool.h>
+#include "adc.h"
+#include "ads1015.h"
+#include "ads1015_def.h"
+#include "delay.h"
+#include "event_arbiter.h"
+#include "event_queue.h"
+#include "gpio_it.h"
+#include "i2c.h"
+#include "input_event.h"
+#include "interrupt.h"
+#include "log.h"
+#include "magnetic_brake_event_generator.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "unity.h"
+
+int16_t percentage_converter(MagneticCalibrationData *data, MagneticBrakeSettings *brake_settings) {
+  int16_t percentage;
+
+  if (brake_settings->zero_value > brake_settings->hundred_value) {
+    percentage =
+        ((brake_settings->min_allowed_range * (data->reading - brake_settings->hundred_value)) /
+         (brake_settings->hundred_value - brake_settings->zero_value)) +
+        brake_settings->max_allowed_range;
+  } else {
+    percentage =
+        (brake_settings->max_allowed_range * (data->reading - brake_settings->zero_value)) /
+        (brake_settings->hundred_value - brake_settings->zero_value);
+  }
+
+  if (percentage < brake_settings->min_allowed_range) {
+    percentage = brake_settings->min_allowed_range;
+  } else if (percentage > brake_settings->max_allowed_range) {
+    percentage = brake_settings->max_allowed_range;
+  }
+
+  // add code to see if the percentage is greater than max allowed value
+  if (percentage > brake_settings->percentage_threshold) {
+    uint16_t percentage_data = (uint16_t)percentage;
+    event_raise(INPUT_EVENT_MECHANICAL_BRAKE_PRESSED, percentage_data);
+  } else {
+    uint16_t percentage_data = (uint16_t)percentage;
+    event_raise(INPUT_EVENT_MECHANICAL_BRAKE_PRESSED, percentage_data);
+  }
+
+  return percentage;
+}
+
+static void input_values(MagneticCalibrationData *data, MagneticBrakeSettings *brake_settings,
+                         Ads1015Channel channel) {
+  int16_t first_samples[1000];
+  int16_t second_samples[1000];
+  int16_t temp1, temp2, temp3, temp4;
+
+  printf("%s\n",
+         "Brake sensor is calibrating, Please ensure the brake is not being pressed, wait \n "
+         "for response to continue");
+
+  delay_s(3);
+
+  for (int i = 0; i < 1000; i++) {
+    // ads1015_read_raw(data->storage, channel, &data->reading);
+    first_samples[i] = data->percentage;
+  }
+
+  // max in range of samples = temp 1 and 3
```

Comment:
What does this refer to? If it is the following block of code you should remove the newlines.